apps/emqx_rule_engine/src/emqx_rule_api_schema.erl

```diff
@@ -307,6 +307,17 @@ fields("ctx_delivery_dropped") ->
         {"from_clientid", sc(binary(), #{desc => ?DESC("event_from_clientid")})},
         {"from_username", sc(binary(), #{desc => ?DESC("event_from_username")})}
         | msg_event_common_fields()
+    ];
+fields("ctx_message_validation_failed") ->
+    Event = 'message.validation_failed',
+    [
+        {"event_type", event_type_sc(Event)},
+        {"clientid", sc(binary(), #{desc => ?DESC("event_clientid")})},
+        {"payload", sc(binary(), #{desc => ?DESC("event_payload")})},
+        qos(),
+        {"topic", sc(binary(), #{desc => ?DESC("event_topic")})},
+        {"username", sc(binary(), #{desc => ?DESC("event_username")})},
+        {"validation", sc(binary(), #{desc => ?DESC("event_validation")})}
```

Comment:
msg_event_common_fields() ?

Reply:
The above fields are the only ones provided by the dashboard when selecting `message_validation_failed`. For example, `peerhost` is in `msg_event_common_fields()` but is not provided by the dashboard so I assume this is not a valid field for the `message_validation_failed` event?

Reply:
which fields the dashboard provide depends on which ones are defined in the schema.

Reply:
:heavy_check_mark: